### PR TITLE
Feature/ggspatial

### PR DESCRIPTION
**Pull request by iskandari**
- replaces `ggmap` with `ggspatial`
- removal of calls to `download_basemap()` 
- addition of `ggspatial` to includes; `rosm` and `prettymapr` to suggests

### CODE REVIEW

File: R/map.R [hunk 1]
```diff
@@ -188,44 +187,51 @@ map.ppi <- function(x, map, param, alpha = 0.7, xlim, ylim, zlim = c(-20, 20),
...
+  rdf <- raster::as.data.frame(r, xy = TRUE)
+  names(rdf) <- c("x", "y", "fill")
+  rdf$fill <- col_func(r@data@values, zlim)
 
   # these declarations prevent generation of NOTE "no visible binding for
   # global variable" during package Check
-  lon <- lat <- y <- z <- NA
+  fill <- lon <- lat <- X <- Y <- y <- z <- NA
   # extract unique radar locations
-  latlon_radar <- unique(data.frame(lat=c(x$geo$lat), lon=c(x$geo$lon)))
```
Comment (adokter) [inline R/map.R:197]: lat and lon attributes can be vectors when using `composite_ppi()`, need to double check compatability with that function. When compositing multiple radars in a ppi, function needs to plot radar locations of all radars

File: vignettes/rad_aero_22.Rmd [hunk 2]
```diff
@@ -156,12 +156,9 @@ plot(my_ppi, param = "VRADH")
...
 ```{r, eval=SHOW_ANSWERS, warning=FALSE}
 # It is often informative to plot radar data on a base layer.
-# first download the background image:
-basemap <- download_basemap(my_ppi)
-# plot the basemap:
-plot(basemap)
+# First choose a base layer from the list of rosm::osm.types()
 # then overlay the PPI on the basemap, restricting the color scale from -20 to 40 dBZ:
-map(my_ppi, map = basemap, param = "DBZH", zlim = c(-20, 40))
+map(my_ppi, basemap = basemap, param = "DBZH", zlim = c(-20, 40))
```
Comment (adokter) [inline vignettes/rad_aero_22.Rmd:161]: keep map for backward compatibility

File: vignettes/range_correction.Rmd [hunk 3]
```diff
@@ -145,8 +145,8 @@ plot(my_corrected_ppi, param = "VIR")
 Or plot the vertically integrated density on a map:
 
 ```{r}
-bm <- download_basemap(my_corrected_ppi)
-map(my_corrected_ppi, bm, param = "VIR", alpha = .5)
+bm <- "osm"
+map(my_corrected_ppi, basemap=bm, param = "VIR", alpha = .5)
```
Comment (adokter) [inline vignettes/range_correction.Rmd:149]: keep map for backward compatibility

File: DESCRIPTION [hunk 4]
```diff
@@ -34,6 +34,7 @@ Imports:
     fields,
     frictionless,
     ggplot2,
+    ggspatial,
```
Comment (adokter) [inline DESCRIPTION:37]: since ggmap was in suggests, can ggspatial be too?

File: DESCRIPTION [hunk 5]
```diff
@@ -53,9 +54,10 @@ Imports:
     viridisLite
 Suggests:
     aws.s3,
-    ggmap (>= 3.0.0),
     knitr,
     rmarkdown,
+    rosm,
+    prettymapr,
```
Comment (adokter) [inline DESCRIPTION:60]: where is prettymapr used? ggspatial imports rosm, so I believe that means they should be either both depends, or both suggests.

I think all should be suggests, and bioRad should be able to run without it. When map is run, we should do a `rlang::check_installed()` test (see example in previous basemap.R) that verifies if ggspatial is installed.

**Review by adokter — CHANGES_REQUESTED** [review]

File: R/composite_ppi.R [hunk 6]
```diff
@@ -74,11 +74,8 @@
 #' # across the available scans at each geographic location
 #' composite <- composite_ppi(ppis, method = "max", res=1000)
 #'
-#' # Download basemap
-#' bm <- download_basemap(composite)
-#'
 #' # Plot the calculated max product on the basemap
-#' map(composite, bm)
+#' map(composite)
```
Comment (adokter) [inline R/composite_ppi.R:78]: map's for composites should so locations for each radar in the composite (I noticed the unit test doesn't test for this functionality)
```
pvol1=read_pvolfile("~/Desktop/KBGM20230726_112800_V06")
pvol2=read_pvolfile("~/Desktop/KHGX20170504_013934_V06")
ppi1=project_as_ppi(pvol1$scans[[1]])
ppi2=project_as_ppi(pvol2$scans[[1]])
ppi3=composite_ppi(list(pp1,pp2))
map(ppi3) # should show two radar locations
```

File: R/bioRad-deprecated.R [hunk 7]
```diff
@@ -40,4 +40,14 @@ vol2bird_version <- function(...) {
...
+#' @section download_basemap:
+#' This function has been deprecated 
+#' ggmap has been replaced by ggspatial which no longer requires a pre-downloaded raster basemap
+#'
+#' @rdname bioRad-deprecated
+#' @export
+#' @return No return value, called for warning message side effect only
+download_basemap <- function(...) {
+  warning("download_basemap has been deprecated; ?bioRad::map for details")
+  return(TRUE)
```
[the remark this answers is not in the document]
Comment (iskandari) [reply R/bioRad-deprecated.R]: Now 0.7.3

**Review by adokter — APPROVED** [review]

**Pull request merged**